moleculegen/base.py

```diff
@@ -1,8 +1,262 @@
 """
 Base objects for all modules.
+
+Classes
+-------
+Token
+    Token enumeration containing possible tokens from SMILES Vocabulary.
+Corpus
+    Descriptor that stores corpus of `Vocabulary` or similar instances.
 """
 
-from typing import Any, List, Optional, Type
+import enum
+from typing import Any, FrozenSet, List, Optional, Type
+
+
+class Token(enum.Enum):
+    """Token enumeration class containing possible tokens from SMILES
+    Vocabulary. Every token entry is an attribute, which can be attained by
+    calling `Token.TOKEN_NAME`. The values of tokens come in form
+    (value, rule_class), in which value is SMILES token and rule class is one
+    of {'atom', 'other', 'special'}. In case of atoms, the names of tokens are
+    identical to the values (atomic symbols), i.e.
+    `Token.TOKEN_NAME.name == Token.TOKEN_NAME.value[0]`. Otherwise, the names
+    are self-explanatory, i.e. `'=' == TOKEN.BOND_DOUBLE.value[0]`.
+    """
```

Comment:
Enumeration is handy for storing, aliasing, searching, and grouping, but I'm afraid a hardcoded Token class with the class attributes corresponding to SMILES rule classes is preferable since enum API seems cumbersome.